docs/languages/powershell.md

```diff
@@ -0,0 +1,132 @@
+
+[Debugging PowerShell script in Visual Studio Code – Part 1](https://blogs.technet.microsoft.com/heyscriptingguy/2017/02/06/debugging-powershell-script-in-visual-studio-code-part-1/)
+
+[Debugging PowerShell script in Visual Studio Code – Part 2](https://blogs.technet.microsoft.com/heyscriptingguy/2017/02/13/debugging-powershell-script-in-visual-studio-code-part-2/)
+
+## Settings Reference
+
+You can customize VSCode settings from the file menu by selecting preferences then clicking settings.
+
+You can also click the gear Icon
+located in the lower left corner of the activity bar.
+
+![codeGear](./images/powershell/codeGear.png)
+
+You can also use the shortcut <kbd>ctrl</kbd>-<kbd>,</kbd> (<kbd>cmd</kbd>-<kbd>,</kbd>) for Mac for a quick search
+
+This will launch the settings.json file, in VSCode versions prior to 1.27.1. , where the VSCode team has introduced a settings GUI, which will be the new default interface for customizing settings.
+
+You can still launch the JSON settings file by using the command __Open Settings (JSON)__ or by changing the default settings editor with the ```"workbench.settings.editor"```
+
+### VSCode Settings GUI
+
+![VSCode Setting GUI](./images/powershell/vscodeSettingsGui.png)
+
+### PowerShell Extension Settings
+
+![PowerShell Extension VSCode Settings](./images/powershell/pwshCodeSettings.png)
+
+## PowerShell Extension multi-version support
+
+You can configure the PowerShell Extensions to use any version of PowerShell installed on your machine by following [these instructions](https://docs.microsoft.com/en-us/powershell/scripting/core-powershell/vscode/using-vscode?view=powershell-6#using-a-specific-installed-version-of-powershell)
+
+## Pester
+
+[Pester](https://github.com/pester/Pester/wiki/Pester)
+ is a Behavior-Driven Development (BDD) based unit test runner for PowerShell.
+Pester provides a framework for running unit tests to execute and validate PowerShell commands. Pester follows a file naming convention for naming tests to be discovered by pester at test time and a simple set of functions that expose a testing DSL for isolating, running, evaluating and reporting the results of PowerShell commands.
+
+Pester 3.40 comes installed with Windows 10 and Server 2016, to update to the latest version follow these instructions from the Pester Github Readme [Pester updated installation instructions](https://github.com/pester/Pester#installation)
+
+## PSScriptAnalyzer
+
+[PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer#introduction) is a static code checker for Windows PowerShell modules and scripts. PSScriptAnalyzer checks the quality of Windows PowerShell code by running a set of rules. The rules are based on PowerShell best practices identified by the PowerShell Team and the community. Pester generates diagnostic results (errors and warnings) to inform users about potential code defects and suggests possible solutions for improvements.
```

Comment:
nit: "Windows PowerShell" should just be "PowerShell" since we have PowerShell Core now 😄